Promote PodOS field to GA (#35590)
kubernetes/website: content/en/docs/concepts/windows/user-guide.md

The recommended approach is outlined below,
with one of its main goals being that this approach should not break compatibility for existing Linux workloads.

> Starting from 1.25, please set `.spec.os.name` for a Pod to indicate the operating system

Reviewer: ```suggestion
Starting from 1.25, you can (and should) set `.spec.os.name` for each Pod, to indicate the operating system
```

Reviewer 2: do we specify what the behavior is if a user doesn't?

Author: Should we specify it in the doc? If we specify, it may encourage users from not setting it, which defeats the purpose.

Reviewer: Ideally, put that detail in as a comment in the upstream comments for the Pod API (if that doesn't land until v1.26, it's OK).

Author: Fair enough, I'll add that info to the pod API docs once the master branch on k/k opens.
